file.go

```diff
@@ -10,14 +14,104 @@ import (
 var (
 	_ fs.File     = (*file)(nil)
 	_ fs.FileInfo = (*fileInfo)(nil)
+	_ io.Seeker   = (*file)(nil)
 )
 
 type file struct {
-	io.ReadCloser
-	stat func() (fs.FileInfo, error)
+	cl     s3iface.S3API
+	bucket string
+	name   string
+
+	realReader      io.ReadCloser
+	stat            func() (fs.FileInfo, error)
+	currentPosition int64
+}
+
+func NewFile(cl s3iface.S3API, bucket string, name string) (fs.File, error) {
```

Comment:
I guess this function doesn't create a file but opens one, doesn't it?

Reply:
This doesn't have to be exported. 

Either revert it and leave it as it was in `Open` 
or
change signature to 
```go
func newFile(cl s3iface.S3API, bucket string, name string) (*file, error)
```

Reply:
right. fixed